taar/recommenders/cache.py

```diff
@@ -0,0 +1,515 @@
+import numpy as np
+import bz2
+import io
+import json
+from google.cloud import storage
+
+from taar.interfaces import IMozLogging, ITAARCache
+
+# taarlite guid guid coinstallation matrix
+COINSTALL_PREFIX = "coinstall|"
+
+# taarlite guid guid coinstallation matrix filtered by
+# minimum installation threshholds
+FILTERED_COINSTALL_PREFIX = "filtered_coinstall|"
+
+# taarlite ranking data
+RANKING_PREFIX = "ranking|"
+
+# taarlite minimum installation threshold
+MIN_INSTALLS_PREFIX = "min_installs|"
+
+# taarlite map of guid->(sum of coinstall counts)
+NORMDATA_COUNT_MAP_PREFIX = "normdata_count_map_prefix|"
+
+# taarlite number of times a GUID shows up per row
+# of coinstallation data.
+NORMDATA_ROWCOUNT_PREFIX = "normdata_rowcount_prefix|"
+
+# taarlite row nownormalization data
+NORMDATA_GUID_ROW_NORM_PREFIX = "normdata_guid_row_norm_prefix|"
+
+# TAAR: Locale data
+LOCALE_DATA = "taar_locale_data|"
+
+# TAAR: collaborative data
+COLLAB_MAPPING_DATA = "taar_collab_mapping|"
+COLLAB_ITEM_MATRIX = "taar_collab_item_matrix|"
+
+# TAAR: similarity data
+SIMILARITY_DONORS = "taar_similarity_donors|"
+SIMILARITY_LRCURVES = "taar_similarity_lrcurves|"
+
+# TAAR: similarity preprocessed data
+SIMILARITY_NUM_DONORS = "taar_similarity_num_donors|"
+SIMILARITY_CONTINUOUS_FEATURES = "taar_similarity_continuous_features|"
+SIMILARITY_CATEGORICAL_FEATURES = "taar_similarity_categorical_features|"
+
+# TAAR: ensemble weights
+
+ENSEMBLE_WEIGHTS = "taar_ensemble_weights|"
+
+# TAAR: whitelist data
+WHITELIST_DATA = "taar_whitelist_data|"
+
+
+class TAARCache(ITAARCache):
```

Comment:
This thing is definitely way more complex than it needs to be, but I tried to touch as little as possible to not break things.